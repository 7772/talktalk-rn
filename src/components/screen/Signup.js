// @flow
import React, { Component } from 'react';
import {
  StyleSheet,
  TouchableOpacity,
  Image,
  Text,
  View,
  ScrollView,
} from 'react-native';

import type {
  ____ViewStyleProp_Internal as ViewStyle,
  ____TextStyleProp_Internal as TextStyle,
  ____ImageStyleProp_Internal as ImageStyle,
} from 'react-native/Libraries/StyleSheet/StyleSheetTypes';

import TextInput from '../shared/TextInput';
import Button from '../shared/Button';
import StatusBar from '../shared/StatusBar';

import { ratio, colors } from '../../utils/Styles';
import { getString } from '../../../STRINGS';

type Styles = {
  container: ViewStyle,
  scrollView: ViewStyle,
  scrollViewContainer: ViewStyle,
  wrapper: ViewStyle,
  btnWrapper: ViewStyle,
  btnRegister: ViewStyle,
  txtRegister: TextStyle,
};

const styles: Styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'white',
    flexDirection: 'column',
    alignItems: 'center',
  },
  scrollView: {
    alignSelf: 'stretch',
  },
  scrollViewContainer: {
    justifyContent: 'center',
    alignItems: 'center',
  },
  wrapper: {
    marginTop: 40,
    width: '78%',

    flexDirection: 'column',
    alignItems: 'center',
  },
  btnWrapper: {
    width: '100%',
    alignItems: 'flex-end',
  },
  btnRegister: {
    backgroundColor: colors.dodgerBlue,
    borderColor: colors.dodgerBlue,
    borderRadius: 4,
    borderWidth: 1,
    width: 136,
    height: 60,
    marginLeft: 4,
    marginTop: 24,
    marginBottom: 48,
    shadowColor: colors.dodgerBlue,
    shadowOffset: {
      width: 0,
      height: 10,
    },
    shadowRadius: 4,
    shadowOpacity: 0.3,

    alignItems: 'center',
    justifyContent: 'center',
  },
  txtRegister: {
    fontSize: 16,
    fontWeight: 'bold',
    color: 'white',
  },
});

type Props = {
  navigation: any;
};
type State = {
  isRegistering: boolean;
  email: string;
  pw: string;
  displayName: string;
  statusMsg: string;
};

class Screen extends Component<Props, State> {
  static navigationOptions = {
    title: getString('SIGNUP'),
  };

  constructor(props: Props) {
    super(props);
    this.state = {
      isRegistering: false,
      email: '',
      pw: '',
      displayName: '',
      statusMsg: '',
    };
  }

  render() {
    return (
      <View style={styles.container}>
        <StatusBar isDarkContent={true}/>
        <ScrollView
          style={styles.scrollView}
          contentContainerStyle={styles.scrollViewContainer}
        >
          <View style={styles.wrapper}>
            <TextInput
              style={{ marginTop: 8 }}
              txtLabel={getString('EMAIL')}
              txtHint={ getString('EMAIL') }
              txt={ this.state.email }
              onTextChanged={ (text) => this.onTextChanged('EMAIL', text)}
            />
            <TextInput
              style={{ marginTop: 24 }}
              txtLabel={ getString('PASSWORD') }
              txtHint={ getString('PASSWORD') }
              txt={ this.state.pw }
              onTextChanged={ (text) => this.onTextChanged('PW', text)}
              isPassword={ true }
            />
            <TextInput
              style={{ marginTop: 24 }}
              txtLabel={getString('NAME')}
              txtHint={ getString('NAME') }
              txt={ this.state.displayName }
              onTextChanged={ (text) => this.onTextChanged('NAME', text)}
            />
            <TextInput
              style={{ marginTop: 24 }}
              txtLabel={getString('STATUS_MSG')}
              txtHint={ getString('STATUS_MSG') }
              txt={ this.state.statusMsg }
              onTextChanged={ (text) => this.onTextChanged('STATUS_MSG', text)}
            />
            <View style={styles.btnWrapper}>
              <Button
                id='register'
                isLoading={this.state.isRegistering}
                onPress={() => this.onRegister()}
                style={styles.btnRegister}
                textStyle={styles.txtRegister}
              >{getString('REGISTER')}</Button>
            </View>
          </View>
        </ScrollView>
      </View>
    );
  }

  onRegister = () => {
    this.setState({ isRegistering: true }, () => {
      this.setState({ isRegistering: false });
    });
  }

  onTextChanged = (type: string, text: string) => {
    switch (type) {
      case 'EMAIL':
        this.setState({ email: text });
        break;
      case 'PW':
        this.setState({ pw: text });
        break;
      case 'NAME':
        this.setState({ displayName: text });
        break;
      case 'STATUS_MSG':
        this.setState({ statusMsg: text });
        break;
    }
  }
}

export default Screen;
