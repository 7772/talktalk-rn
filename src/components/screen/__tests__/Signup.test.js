import 'react-native';
import * as React from 'react';
import Signup from '../Signup';
// import appStore from '../../../stores/appStore';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { shallow, render } from 'enzyme';

describe('rendering test', () => {
  const wrapper = shallow(
    <Signup />,
    // <Signup store={appStore}/>,
  );

  it('renders as expected', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
