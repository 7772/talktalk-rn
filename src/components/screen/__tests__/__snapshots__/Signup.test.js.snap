// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering test renders as expected 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <Shared
    isDarkContent={true}
  />
  <ScrollView
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
    style={
      Object {
        "alignSelf": "stretch",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "marginTop": 40,
          "width": "78%",
        }
      }
    >
      <Shared
        labelStyle={
          Object {
            "color": "rgb(134,154,183)",
            "fontSize": 12,
            "marginBottom": 8,
          }
        }
        labelStyleFocus={
          Object {
            "color": "rgb(58,139,255)",
            "fontSize": 12,
            "marginBottom": 8,
          }
        }
        onTextChanged={[Function]}
        style={
          Object {
            "marginTop": 8,
          }
        }
        txt=""
        txtHint="Email"
        txtLabel="Email"
      />
      <Shared
        isPassword={true}
        labelStyle={
          Object {
            "color": "rgb(134,154,183)",
            "fontSize": 12,
            "marginBottom": 8,
          }
        }
        labelStyleFocus={
          Object {
            "color": "rgb(58,139,255)",
            "fontSize": 12,
            "marginBottom": 8,
          }
        }
        onTextChanged={[Function]}
        style={
          Object {
            "marginTop": 24,
          }
        }
        txt=""
        txtHint="Password"
        txtLabel="Password"
      />
      <Shared
        labelStyle={
          Object {
            "color": "rgb(134,154,183)",
            "fontSize": 12,
            "marginBottom": 8,
          }
        }
        labelStyleFocus={
          Object {
            "color": "rgb(58,139,255)",
            "fontSize": 12,
            "marginBottom": 8,
          }
        }
        onTextChanged={[Function]}
        style={
          Object {
            "marginTop": 24,
          }
        }
        txt=""
        txtHint="NAME"
        txtLabel="NAME"
      />
      <Shared
        labelStyle={
          Object {
            "color": "rgb(134,154,183)",
            "fontSize": 12,
            "marginBottom": 8,
          }
        }
        labelStyleFocus={
          Object {
            "color": "rgb(58,139,255)",
            "fontSize": 12,
            "marginBottom": 8,
          }
        }
        onTextChanged={[Function]}
        style={
          Object {
            "marginTop": 24,
          }
        }
        txt=""
        txtHint="Status message"
        txtLabel="Status message"
      />
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "width": "100%",
          }
        }
      >
        <Button
          activeOpacity={0.5}
          id="register"
          imgLeftStyle={
            Object {
              "height": 24,
              "left": 16,
              "position": "absolute",
              "width": 24,
            }
          }
          indicatorColor="white"
          isDisabled={false}
          isLoading={false}
          onPress={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgb(58,139,255)",
              "borderColor": "rgb(58,139,255)",
              "borderRadius": 4,
              "borderWidth": 1,
              "height": 60,
              "justifyContent": "center",
              "marginBottom": 48,
              "marginLeft": 4,
              "marginTop": 24,
              "shadowColor": "rgb(58,139,255)",
              "shadowOffset": Object {
                "height": 10,
                "width": 0,
              },
              "shadowOpacity": 0.3,
              "shadowRadius": 4,
              "width": 136,
            }
          }
          textStyle={
            Object {
              "color": "white",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          REGISTER
        </Button>
      </View>
    </View>
  </ScrollView>
</View>
`;
